### PR TITLE
refactor request_options, add language param

### PR DESCRIPTION
**Pull request by jacalata**
I have refactored the classes to separate options that can be used in querying content, and options that can be used for exporting data. 

In the new Data Exports section, I've added the parameter "language" which can be used to change how the data in a workbook is localized for export.

### CODE REVIEW

File: tableauserverclient/server/request_options.py [hunk 1]
```diff
@@ -117,51 +156,47 @@ class Direction:
...
+"""
 
-    def page_size(self, page_size):
-        self.pagesize = page_size
-        return self
 
-    def page_number(self, page_number):
-        self.pagenumber = page_number
-        return self
+class _DataExportOptions(RequestOptionsBase):
```
Comment (renoyjohnm) [inline tableauserverclient/server/request_options.py:165]: nit: is `_VizExportOptions` a better class name here as I believe options like `maxAge` are limited to exporting API endpoints

File: tableauserverclient/server/request_options.py [hunk 2]
```diff
@@ -182,82 +217,33 @@ def _append_view_filters(self, params) -> None:
...
+    extension = "png"
 
-class ImageRequestOptions(_FilterOptionsBase):
     # if 'high' isn't specified, the REST API endpoint returns an image with standard resolution
     class Resolution:
         High = "high"
 
     def __init__(self, imageresolution=None, maxage=-1):
-        super().__init__()
+        super().__init__(maxage=maxage)
```
Comment (renoyjohnm) [inline tableauserverclient/server/request_options.py:236]: may be a good time to add `viz_height` & `viz_width` to image request options?

**Review by renoyjohnm — APPROVED** [review]

File: tableauserverclient/server/request_options.py [hunk 3]
```diff
@@ -117,51 +156,47 @@ class Direction:
...
+    @property_is_int(range=(0, 240), allowed=[-1])
+    def max_age(self, value):
+        self._max_age = value
 
     def get_query_params(self):
-        raise NotImplementedError()
+        params = {}
+        if self.max_age != -1:
+            params["maxAge"] = self.max_age
+        if self.language:
```
Comment (anyoung-tableau) [inline tableauserverclient/server/request_options.py:195]: Should this be `self._language`? I only see a setter for `language`, not a getter.

File: tableauserverclient/server/request_options.py [hunk 4]
```diff
@@ -117,51 +156,47 @@ class Direction:
...
-        if self._all_fields:
-            params["fields"] = "_all_"
-        return params
+    """
+    This setting will affect the contents of the workbook as they are exported.
+    Valid language values are tableau-supported languages like de, es, en
+    If no locale is specified, the default locale for that language will be used
+    """
 
+    def language(self, value):
```
Comment (anyoung-tableau) [inline tableauserverclient/server/request_options.py:179]: Does this need a `@language.setter` decorator?

File: tableauserverclient/server/request_options.py [hunk 5]
```diff
@@ -117,51 +156,52 @@ class Direction:
...
+    def max_age(self, value):
+        self._max_age = value
 
     def get_query_params(self):
-        raise NotImplementedError()
+        params = {}
+        if self.max_age != -1:
+            params["maxAge"] = self.max_age
+        if self.language:
+            params["language"] = self._language
```
Comment (anyoung-tableau) [inline tableauserverclient/server/request_options.py:201]:
```suggestion
            params["language"] = self.language
```

File: samples/export.py [hunk 6]
```diff
@@ -74,16 +74,18 @@ def main():
...
+        options: TSC.PDFRequestOptions = option_factory()
 
         if args.filter:
-            options = option_factory().vf(*args.filter.split(":"))
-        else:
-            options = None
+            options = options.vf(*args.filter.split(":"))
+
+        if not options.language:
+            options.language = "de"
```
[the remark this answers is not in the document]
Comment (jacalata) [reply samples/export.py]: yep, removed and added help detail to the option itself.

**Pull request merged**